pkg/proxy/proxy.go

```diff
@@ -200,11 +206,16 @@ func doTokenRequest(ctx context.Context, clientID, resource, tenantID, authority
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to acquire token")
 	}
+	expiresIn, err := parseExpiresIn(strconv.FormatInt(result.ExpiresOn.UTC().Unix()-time.Now().UTC().Unix(), 10))
```

Comment:
This is susceptible to clock skew between the token creator and the proxy's current time.